.github/workflows/cd-prepare.yml

```diff
@@ -110,53 +112,59 @@ jobs:
 
             # If the components.json file, so it seems has external packages
             if [[ -f "$components_json_file" ]]; then
-              declare -A packages=()
-              # Iterating over the external components to get the involved elements
-              while read -r component_id; do
-                if [[ -n "$component_id" ]]; then
-                  component_version_full=$(jq -cr '."'${component_id}'".version' "$components_json_file")
+              while read -r architecture; do
+                # Remove platform prefix.
+                arch="${architecture##*/}"
+                declare -A packages=()
+                # Iterating over the external components to get the involved elements
+                while read -r component_id; do
+                  if [[ -n "$component_id" ]]; then
+                    component_version_full=$(jq -cr '."'${component_id}'".version' "$components_json_file")
 
-                  #split by "-"
-                  component_parts=(${component_version_full//-/ })
-                  component_version=${component_parts[0]}
-                  component_distro=$(jq -cr '."'${component_id}'".distro' "$components_json_file")
-                  component_arch=$(jq -cr '."'${component_id}'".arch' "$components_json_file")
-                  # Added "true" to avoid errors if compilation_recipe doesn't exists
-                  compilation_recipe=$(s3cmd ls -l "s3://${AWS_S3_BUCKET}/$component_id/$component_version/$component_distro/$component_arch/" | grep -c "compilation-recipe.json" || true)
-                  # If the components.json file, so it seems has external packages
-                  if [[ $compilation_recipe -gt 0 ]]; then
-                    s3cmd get  "s3://${AWS_S3_BUCKET}/$component_id/$component_version/$component_distro/$component_arch/compilation-recipe.json" compilation-recipe.json
-                  else
-                    s3cmd get "s3://${AWS_S3_BUCKET}/$component_id/$component_version-${component_parts[1]}/$component_distro/$component_arch/compilation-recipe.json" compilation-recipe.json
+                    #split by "-"
+                    component_parts=(${component_version_full//-/ })
+                    component_version=${component_parts[0]}
+                    component_distro=$(jq -cr '."'${component_id}'".distro' "$components_json_file")
+                    # We will use the container arch instead of the component arch (component file has only amd64 references)
+                    # component_arch=$(jq -cr '."'${component_id}'".arch' "$components_json_file")
+                    component_arch=$arch
```

Comment:
Current `components.json` file contains references to amd64 achitecture by default. We will use the container architecture to get the recipe.